ports/webassembly/proxy_js.js

```diff
@@ -97,13 +98,19 @@ function proxy_js_check_existing(c_ref) {
 
 // js_obj cannot be undefined
 function proxy_js_add_obj(js_obj) {
+    // See if there is an existing JsProxy reference, and use that if there is.
+    if (proxy_js_ref_map.has(js_obj)) {
```

Comment:
I know it's an ugly workaround but AFAIK there are no optimizations for `has` immediately followed by `get` in the JS *Map* world (runtimes) so that when performance are meant to be "best" and/or on the critical path, *JS* devs tend to abuse `undefined` returned when no entry is found.

That is:

```js
const known = proxy_js_ref_map.get(js_obj);
if (known) return known;
```

We're still talking *O(1)* with JS *Map* lookup for the key but while *O(1)* + *O(1)* is still *O(1)* in the theoretical side of *BigO* notation, you can skip that "*O(2)*" by reaching directly the value, if available, via `.get(key)`.

Not a blocker, of course, but years of perf tuning makes me always use that ugly *fast-path* whenever it's crucial.

Reply:
yes, that's the falsy value I was referring to ... if `0` could be returned, you definitively need that `!== undefined` check 👍